lib/cc/engine/analyzers/reporter.rb

```diff
@@ -64,9 +64,17 @@ def mass_threshold
           @mass_threshold ||= language_strategy.mass_threshold
         end
 
+        def base_points
+          @base_points ||= language_strategy.base_points
+        end
+
+        def points_per
+          @points_per ||= language_strategy.points_per
+        end
+
         def new_violation(issue)
           hashes = flay.hashes[issue.structural_hash]
-          Violation.new(language_strategy.base_points, issue, hashes)
+          Violation.new(base_points, points_per, mass_threshold, issue, hashes)
```

Comment:
@wfleming Yeah, I think I agree with you.

However, I think that's a separate refactor PR. I think this PR should stay focused on the points changes to minimize impact, but open to other thoughts.